File: JAVA/BasicForProgramming/BinToDcl.java
import java.util.*;
import java.io.*;

public class BinToDcl {
	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);

		int n = in.nextInt();
		int no = 0, p = 1;
		int temp = n;

		while (temp > 0) {
			int dgt = temp % 10;
			no += dgt * p;
			p *= 2;
			temp /= 10;
		}

		System.out.println(no);
		dcltobin(n);
	}

	static void dcltobin(int n) {

		while (n > 0) {
			System.out.print(n % 2);
			n = n / 2;
		}
		System.out.println();
	}
}

File: CPP/BasicForProgramming/bintodcl.cpp
#include <iostream>
#include <cmath>

using namespace std;

void dcltobin(int n) {

	while (n > 0) {
		int r = n % 2;
		cout << r;
		n /= 2;
	}
	cout << endl;
}

int main() {
	long long int n;  cin >> n;
	long long int temp = n;
	int no = 0;
	int p = 1;

	while (temp > 0) {
		int dgt = temp % 10;
		no += dgt * p;
		p *= 2;
		temp /= 10;
	}
	cout << no << endl;
	dcltobin(no);
}